cpp/ql/test/library-tests/controlflow/nullness/nullness.expected

```diff
@@ -7,9 +7,14 @@
 | test.cpp:15:8:15:23 | call to __builtin_expect | test.cpp:5:13:5:13 | v | is not null | is valid |
 | test.cpp:16:8:16:23 | call to __builtin_expect | test.cpp:5:13:5:13 | v | is null | is not valid |
 | test.cpp:17:9:17:17 | ... && ... | test.cpp:5:13:5:13 | v | is not null | is valid |
-| test.cpp:18:9:18:17 | ... && ... | test.cpp:5:13:5:13 | v | is not null | is not valid |
+| test.cpp:18:9:18:17 | ... && ... | test.cpp:5:13:5:13 | v | is not null | is valid |
 | test.cpp:19:9:19:18 | ... && ... | test.cpp:5:13:5:13 | v | is null | is not valid |
-| test.cpp:20:9:20:18 | ... && ... | test.cpp:5:13:5:13 | v | is not null | is not valid |
+| test.cpp:20:9:20:18 | ... && ... | test.cpp:5:13:5:13 | v | is null | is not valid |
 | test.cpp:21:9:21:14 | ... = ... | test.cpp:5:13:5:13 | v | is null | is not valid |
 | test.cpp:21:9:21:14 | ... = ... | test.cpp:7:10:7:10 | b | is not null | is valid |
-| test.cpp:22:17:22:17 | b | test.cpp:7:10:7:10 | b | is not null | is valid |
+| test.cpp:22:9:22:14 | ... = ... | test.cpp:5:13:5:13 | v | is not null | is not valid |
+| test.cpp:22:9:22:14 | ... = ... | test.cpp:7:13:7:13 | c | is not null | is not valid |
```

Comment:
Please would you explain these two results?  It looks to me like if `c = !v` is true, `c` will be true (only after the assignment) and `v` must be false.

If this is just a limitation of the analysis, I'm happy with that and happy for it to be documented by this test.  I just want to check I'm understanding correctly.

Reply:
Yeah, those are a bit weird. There's two things going on here:

1. RHSs of assignments are not covered, as they're technically not part of a condition. So, those expressions for each variable will always result in `is not null` and `is not valid`
2. The `c` case is a C++17-style if-initializer (note the `c` in the condition on the next line of the expected file). These also fall outside the condition that is evaluated in the `if`, and hence we also get `is not null` and `is not valid` there.